tests/debugging/test_encoding.py

```diff
@@ -265,3 +265,87 @@ def test_format_message(args, expected):
 
 def test_encoding_none():
     assert utils.capture_value(None) == {"isNull": True, "type": "NoneType"}
+
+
+@pytest.mark.parametrize(
+    "count,result",
+    [
+        (1, {"entries": [], "notCapturedReason": "CountBudget", "size": 1, "type": "dict"}),
+        (
+            5,
+            {
+                "type": "dict",
+                "entries": [
+                    (
+                        {"type": "str", "value": "'a'"},
+                        {
+                            "type": "dict",
+                            "entries": [
+                                (
+                                    {"type": "str", "notCapturedReason": "CountBudget"},
+                                    {"type": "dict", "notCapturedReason": "CountBudget", "size": 1},
+                                )
+                            ],
+                            "size": 1,
+                        },
+                    )
+                ],
+                "size": 1,
+            },
+        ),
+        (
+            10,
+            {
+                "type": "dict",
+                "entries": [
+                    (
+                        {"type": "str", "value": "'a'"},
+                        {
+                            "type": "dict",
+                            "entries": [
+                                (
+                                    {"type": "str", "value": "'a'"},
+                                    {
+                                        "type": "dict",
+                                        "entries": [
+                                            (
+                                                {"type": "str", "value": "'a'"},
+                                                {
+                                                    "type": "dict",
+                                                    "entries": [],
+                                                    "size": 1,
+                                                    "notCapturedReason": "CountBudget",
+                                                },
+                                            )
+                                        ],
+                                        "size": 1,
+                                    },
+                                )
+                            ],
+                            "size": 1,
+                        },
+                    )
+                ],
+                "size": 1,
+            },
+        ),
+    ],
+)
+def test_encoding_stopping_cond(count, result):
```

Comment:
Do we want validate all other edge cases (too many fields, to many items, timeout in the middle of a collection) like we did for serialize function?

Maybe another ticket?

Reply:
adding